sunxspex/thermal.py

```diff
@@ -0,0 +1,699 @@
+        URL or file location of the CHIANTI IDL save file to be used.
+    """
+    global _LINE_GRID
+    if filename:
+        with manager.override_file("chianti_lines", uri=filename):
+            line_info = load_chianti_lines_lite()
+    else:
+        line_info = load_chianti_lines_lite()
+    _LINE_GRID = {}
+    _LINE_GRID["intensity"] = np.array(line_info.data)
+    _LINE_GRID["intensity unit"] = line_info.attrs["units"]["data"]
+    _LINE_GRID["intensity description"] = (
+        "Intensity is stored as photons per unit emission measure at the source.  "
+        "It (and its unit) therefore must be multipled by emission measure and "
+        "divided by 4 * pi * observer_distance**2 to get observed values.")
+    _LINE_GRID["line_peaks_keV"] = (
+            line_info.peak_energy.data * line_info.attrs["units"]["peak_energy"]).to_value(
+                u.keV, equivalencies=u.spectral())
+    _LINE_GRID["log10T"] = line_info.logT.data
+    _LINE_GRID["element index"] = line_info.attrs["element_index"]
+    _LINE_GRID["line atomic numbers"] = line_info.atomic_number.data
+
+
+def setup_default_abundances(filename=None):
+    """
+    Read default abundance values into global variable.
+
+    By default, data is read from the following file:
+    https://hesperia.gsfc.nasa.gov/ssw/packages/xray/dbase/chianti/xray_abun_file.genx
+    To load data from a different file, see Notes section.
+
+    Parameters
+    ----------
+    filename: `str` (optional)
+        URL or file location of the .genx abundance file to be used.
+    """
+    global DEFAULT_ABUNDANCES
+    if filename:
+        with manager.override_file("xray_abundance", uri=filename):
+            DEFAULT_ABUNDANCES = load_xray_abundances()
+    else:
+        DEFAULT_ABUNDANCES = load_xray_abundances()
+
+# Read line, continuum and abundance data into global variables.
+setup_continuum_parameters()
+setup_line_parameters()
+setup_default_abundances()
+
+
+@u.quantity_input(energy_edges=u.keV,
+                  temperature=u.K,
+                  emission_measure=(u.cm**(-3), u.cm**(-5)),
+                  observer_distance=u.cm)
+def thermal_emission(energy_edges,
+                     temperature,
+                     emission_measure,
+                     abundance_type="sun_coronal",
+                     relative_abundances=None,
+                     observer_distance=(1*u.AU).to(u.cm)):
+    f"""Calculate the thermal X-ray emission (lines + continuum) from the solar atmosphere.
+
+    The emission is calculated as a function of temperature and emission measure.
+    Which continuum mechanisms are included --- free-free, free-bound, or two-photon --- are
+    determined by the file from which the comtinuum parameters are loaded.
+    To change the file used, see the setup_continuum_parameters() function.
+    
+    {doc_string_params}"""
+    energy_edges_keV = energy_edges.to_value(u.keV)
+    temperature_K = temperature.to_value(u.K)
```

Comment:
Using that equivalency is actually quite computationally slow.  It may be better to force people to use a traditional temperature unit.  Also, usiing the equivalency leaves more scope for people to make mistakes.  For example, if they enter the energy and temperature inputs the wrong way around, this equivalency could stop the code erroring when it should.  What do you think?